src/com/opc/opcconnectionimpl.cpp

```diff
@@ -236,8 +236,7 @@ int COpcConnectionImpl::sendItemData(const char *paGroupName, const char *paItem
   if(it != mOpcItems.end()){
     items = it->second;
     for(size_t i = 0; i < items.size(); i++){
-      std::string itemName = WS2S(items[i]->getName());
-      if(0 == strcmp(itemName.c_str(), paItemName)){
+      if(0 == strcmp(WS2S(items[i]->getName()).c_str(), paItemName)){
```

Comment:
as you are comparing a std::string with a char * you can just use:
`if(WS2S(items[i]->getName()) == paItemName) {` 

I guess it would also make sense to consider change paItemName to a std:string, but I would see this in a separate PR.

Reply:
Yes, I just restore the previous initial update. Save it as a temporary std::string not needed.

Reply:
@kumajaya afaik c_str() generates a C-String representation which is then used for strcmp, while using the comparison operator should directly compare to the internal storage representation of std::string without creating the intermediate C-String representation.

Reply:
Furthermore std::string is memory save and the source is mostly also std::string. Converting back and forth is unnecessary and potential dangerous overhead.

Reply:
Thank you for the explanation. Currently I'm working on code refactoring from char* to std::string. More c_sting() conversion on DEVLOG but core logic string operation much simple.

Reply:
DEVLOG is definitly an issue that we also need to look into. But if it makes the rest of to OPC code better I would go for that.

Reply:
> DEVLOG is definitly an issue that we also need to look into. But if it makes the rest of to OPC code better I would go for that.

Sir, may I just push the changes here as another commit?

Reply:
I guess an own PR would be nicer or?

Reply:
> I guess an own PR would be nicer or?

Actually yes, but it's depend on this PR.

Reply:
Ok then it makes more sense that you push it on top of this PR and I will not squash the changes.